brave/brave-core
# fix(wallet): limit account name character length

Resolves https://github.com/brave/brave-browser/issues/23361
Resolves https://github.com/brave/brave-browser/issues/37362



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

Scenario 1 (Create Account)
1. Go to accounts page
2. Click the "+" icon
3. click "Create account"
4. Select a coin type
5. enter an account name over 30 characters long
6. an error message should be shown and the submit button should be disabled until the name is 30 characters or shorter
7. Submit button should continue to work (creates an account with the supplied name)
8. repeat for all coin types

Scenario 2 (Import Account)
1. Go to accounts page
2. Click the "+" icon
3. click "Import account"
4. Select a coin type
5. enter an account name over 30 characters long
6. an error message should be shown and the submit button should be disabled until the name is 30 characters or shorter
7. enter a private key
8. Submit button should continue to work (imports the private key as an account with the supplied name)
9. repeat for all coin types
10. repeat using a JSON keystore file instead of a pasted private key (Import type)

Scenario 3 (Edit Account)
1. Go to accounts page
2. Click the kabab icon on an account
3. click "Edit"
4. enter an account name over 30 characters long
5. an error message should be shown and the submit button should be disabled until the name is 30 characters or shorter
6. Submit button should continue to work (edits the account to use the newly supplied name)

Scenario 4 (View Private key)
1. Go to accounts page
2. Click the kabab icon on an account
3. click "Export"
4. enter you wallet password
5. click "show key"
6. private key should be revealed (should not be revealed if password is incorrect, should lock wallet after 3 incorrect attempts)

Scenario 4 (Remove account)
1. Go to accounts page
2. Click the kabab icon on an imported account that can be removed
3. click "Remove"
4. enter you wallet password
5. click "continue"
6. the account should be removed (should not be removed if password is incorrect, should lock wallet after 3 incorrect attempts)
